packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart

```diff
@@ -456,7 +500,7 @@ class FuchsiaDevice extends Device {
 
     // In a debug or profile build, try to find the observatory uri.
     final FuchsiaIsolateDiscoveryProtocol discovery =
-      getIsolateDiscoveryProtocol(appName);
+      getIsolateDiscoveryProtocol(fuchsiaUrl);
```

Comment:
I don't understand this part of the change. What is the reason for changing this from the app name to the full url? Is there a unit test that can cover this change?

Reply:
This is actually part of the change in observatory discovery. This should not be included in this patch. Previously the Observatory discovery prototool identify the app we are running by connecting to vmservice of each running app component and then compare each name of flutter isolaten with appName, the new implementation does all discovery based on /hub and comparing the component url. More explanation are written in #85450.

Right now I should stick with appName on this PR. Let me change it soon.

Reply:
@zanderso Please review again. I have revertd those changes.